Sources/web3swift/Utils/EIP/EIP712/EIP712Parser.swift

```diff
@@ -0,0 +1,297 @@
+
+        var _coreType = self.type
+        if _coreType.hasSuffix("[]") {
+            _coreType.removeLast(2)
+            isArray = true
+        } else {
+            isArray = false
+        }
+        self.coreType = _coreType
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        let name = try container.decode(String.self, forKey: .name)
+        let type = try container.decode(String.self, forKey: .type)
+        self.init(name: name, type: type)
+    }
+}
+
+public struct EIP712TypedData {
+    public let types: [String: [EIP712TypeProperty]]
+    /// A name of one of the types from `types`.
+    public let primaryType: String
+    /// A JSON object as a string
+    public let domain: [String : AnyObject]
+    /// A JSON object as a string
+    public let message: [String : AnyObject]
+
+    public init(types: [String : [EIP712TypeProperty]],
+                primaryType: String,
+                domain: [String : AnyObject],
+                message: [String : AnyObject]) throws {
+        self.types = types
+        self.primaryType = primaryType.trimmingCharacters(in: .whitespacesAndNewlines)
+        self.domain = domain
+        self.message = message
+    }
+
+    public func encodeType(_ type: String) throws -> String {
+        guard let typeData = types[type] else {
+            throw Web3Error.processingError(desc: "EIP712Parser. Attempting to encode type that doesn't exist in this payload. Given type: \(type). Available types: \(types.keys).")
+        }
+        return try encodeType(type, typeData)
+    }
+
+    public func typeHash(_ type: String) throws -> String {
+        try encodeType(type).sha3(.keccak256).addHexPrefix()
+    }
+
+    internal func encodeType(_ type: String, _ typeData: [EIP712TypeProperty], typesCovered: [String] = []) throws -> String {
+        var typesCovered = typesCovered
+        var encodedSubtypes: [String : String] = [:]
+        let parameters = try typeData.map { attributeType in
+            if let innerTypes = types[attributeType.coreType], !typesCovered.contains(attributeType.coreType) {
+                typesCovered.append(attributeType.coreType)
+                if attributeType.coreType != type {
+                    encodedSubtypes[attributeType.coreType] = try encodeType(attributeType.coreType, innerTypes)
+                }
+            }
+            return "\(attributeType.type) \(attributeType.name)"
+        }
+        return type + "(" + parameters.joined(separator: ",") + ")" + encodedSubtypes.sorted { lhs, rhs in
+            return lhs.key < rhs.key
+        }
+        .map { $0.value }
+        .joined(separator: "")
+    }
+
+    /// Convenience function for ``encodeData(_:data:)`` that uses ``primaryType`` and ``message`` as values.
+    /// - Returns: encoded data based on ``primaryType`` and ``message``.
+    public func encodeData() throws -> Data {
```

Comment:
I wonder why this is a one of a two public methods within the class that are having DocC comments? Shouldn't we add some least to other public one?